packages-ts/gauntlet-terra-contracts/src/commands/contracts/migration/index.ts

```diff
@@ -0,0 +1,80 @@
+import { Result } from '@chainlink/gauntlet-core'
+import { prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { TerraCommand, TransactionResponse } from '@chainlink/gauntlet-terra'
+import { AccAddress, MsgMigrateContract } from '@terra-money/terra.js'
+
+type CommandInput<MsgMigrate> = {
+  contract: string
+  newCodeId: number
+  migrationMsg: MsgMigrate
+}
+
+export type MigrationMessageMaker<MsgMigrate> = (flags, args) => MsgMigrate
+
+export abstract class MigrateContract<MsgMigrate> extends TerraCommand {
+  static makeDescription = (contract) =>
+    `(NOT TESTED!) Upgrades a contract ${contract} instance to use a new Code ID. This new Code ID must expose a "migrate" function to migrate the contract state`
+  static makeExamples = (contract) => [
+    `yarn gauntlet ${contract}:migrate_contract --network=<NETWORK> --newCodeId=<CODE_ID_NUMBER> <CONTRACT_ADDRESS>`,
+    `yarn gauntlet ${contract}:migrate_contract --network=mainnet --newCodeId=2012 terra167ccv2h0z7k0p8j6qpuzwsgu5au5qvfwgmkjsl`,
+  ]
+
+  static makeId = (contract) => `${contract}:migrate_contract`
+  static makeCategory = (contract) => contract
+
+  abstract makeMigrationMessage: MigrationMessageMaker<MsgMigrate>
+  input: CommandInput<MsgMigrate>
+
+  constructor(flags, args: string[]) {
+    super(flags, args)
+  }
+
+  buildCommand = async (flags, args): Promise<TerraCommand> => {
+    this.input = this.makeInput(flags, args)
+    this.validateInput(this.input)
+    return this
+  }
+
+  beforeExecute = async () => {
+    await prompt(`Continue upgrading contract ${this.input.contract} to new Code ID ${this.input.newCodeId}?`)
+  }
+
+  makeInput = (flags, args): CommandInput<MsgMigrate> => {
+    return {
+      newCodeId: Number(flags.newCodeId),
+      contract: args[0],
+      migrationMsg: this.makeMigrationMessage(flags, args),
+    }
+  }
+
+  validateInput = (input: CommandInput<MsgMigrate>): boolean => {
```

Comment:
Consider changing this to:

```suggestion
  requireValidInput = (input: CommandInput<MsgMigrate>) => {
```

as this is more explicit to how it's currently used.